base/src/commonMain/kotlin/utils/TemperatureMeasurement.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base.utils
+
+import com.splendo.kaluga.base.text.NumberFormatStyle
+import com.splendo.kaluga.base.text.NumberFormatter
+
+class TemperatureMeasurement(
+    val value: Double,
+    val unit: TemperatureUnit
+) {
+
+    fun convert(toUnit: TemperatureUnit) = when (Pair(unit, toUnit)) {
```

Comment:
Unfortunately kotlin does not have pattern matching or destructing, so the safe way to write this (to trigger warnings when new types are added) is a when inside a when.

Another approach altogether is that `Fahrenheit` and `Celsius` are typed (inline) classes that explicitly define what they can be converted to (e.g. `Celsius` could have a `.fahrenheit` property. This could even be defined in a `Temperature` interface. This will make handling a value typesafe (so you can't accidentally forget to do a conversion from Fahrenheit to Celsius), and due to inline classes there is better performance.

I would favour this approach.